conf/index.html

```diff
@@ -0,0 +1,110 @@
+    <title>TrueNAS Installer</title>
+</head>
+
+<body>
+    <div id="status">Connecting...</div>
+    <div id="messages"></div>
+
+    <script>
+        class RPCClient {
+            constructor(url) {
+                this.ws = new WebSocket(url);
+                this.pendingRequests = new Map();
+                this.setupEventHandlers();
+            }
+
+            setupEventHandlers() {
+                this.ws.onopen = (event) => {
+                    document.getElementById('status').textContent = 'Connected';
+                    console.log('WebSocket connected', event);
+                    this.onConnected();
+                };
+
+                this.ws.onmessage = (event) => {
+                    try {
+                        const response = JSON.parse(event.data);
+                        this.handleResponse(response);
+                    } catch (error) {
+                        console.error('Failed to parse response:', error);
+                    }
+                };
+
+                this.ws.onclose = (event) => {
+                    document.getElementById('status').textContent = 'Disconnected';
+                    console.log('WebSocket closed');
+                };
+
+                this.ws.onerror = (error) => {
+                    document.getElementById('status').textContent = 'Error';
+                    console.error('WebSocket error:', error);
+                };
+            }
+
+            call(method, params = []) {
+                return new Promise((resolve, reject) => {
+                    const id = crypto.randomUUID();
+                    const request = {
+                        id,
+                        jsonrpc: '2.0',
+                        method,
+                        params
+                    };
+
+                    this.pendingRequests.set(id, { resolve, reject });
+                    this.ws.send(JSON.stringify(request));
+                });
+            }
+
+            handleResponse(response) {
+                const messages = document.getElementById('messages');
+                const messageElement = document.createElement('div');
+                messageElement.textContent = JSON.stringify(response, null, 2);
+                messages.appendChild(messageElement);
+                console.log('Message received:', response);
+
+                if (response.id && this.pendingRequests.has(response.id)) {
+                    const { resolve, reject } = this.pendingRequests.get(response.id);
+                    this.pendingRequests.delete(response.id);
+
+                    if (response.error) {
+                        reject(response.error);
+                    } else {
+                        resolve(response.result);
+                    }
+                }
+            }
+
+            async onConnected() {
+                try {
+                    const adoptResult = await this.call('adopt');
```

Comment:
Is there going to be an issue when this script adopts, then TNC tries to authenticate after registration and cert generation on the installer?